crates/cli/src/dirty_git_check.rs

```diff
@@ -0,0 +1,28 @@
+use std::io::{self, Write};
+
+use anyhow::Result;
+use std::process::Command as ProcessCommand;
+
+pub fn dirt_check(allow_dirty: bool) -> Result<()> {
+    if !allow_dirty {
+        let output = ProcessCommand::new("git")
+            .args(["status", "--porcelain"])
+            .output()
+            .expect("Failed to run git");
```

Comment:
A few things:
- The user may not have `git` installed.
- The dir may not be a git dir (git is not initialized in that dir).

In these cases, we should ignore this check